### ISSUE TITLE
Hercules is not listening for CNSLPORT

### ISSUE THREAD

**Issue by miguemely** [SDL-Hercules-390/hyperion#692]
When using hercules, no matter with a valid configuration or `hercules -f none` and defining `CNSLPORT 23`, it looks like it doesn't actually listen on the interface based on the output of `netstat -anop`

This was built using https://github.com/wrljet/hercules-helper, 

Version output:
```
HHC01413I Hercules version 4.8.0.11356-SDL-DEV-ga215caf4
HHC01414I (C) Copyright 1999-2024 by Roger Bowler, Jan Jaeger, and others
HHC01417I Built for you with Hercules-Helper (version: -unknown-)
HHC01415I Build date: Nov 22 2024 at 13:43:52
HHC01417I Built with: GCC 12.2.0
HHC01417I Build type: GNU/Linux x86_64 host architecture build
HHC01417I Modes: S/370 ESA/390 z/Arch
HHC01417I Max CPU Engines: 128
HHC01417I Using   shared libraries
HHC01417I Using   setresuid() for setting privileges
HHC01417I Using   POSIX threads Threading Model
HHC01417I Using   Error-Checking Mutex Locking Model
HHC01417I With    Shared Devices support
HHC01417I With    Dynamic loading support
HHC01417I With    External GUI support
HHC01417I With    IPV6 support
HHC01417I With    HTTP Server support
HHC01417I With    sqrtl support
HHC01417I With    Signal handling
HHC01417I With    Watchdog monitoring
HHC01417I With    CCKD BZIP2 support
HHC01417I With    HET BZIP2 support
HHC01417I With    ZLIB support
HHC01417I With    Regular Expressions support
HHC01417I Without Object REXX support
HHC01417I With    Regina REXX support
HHC01417I With    Automatic Operator support
HHC01417I Without National Language Support
HHC01417I With    CCKD64 Support
HHC01417I With    Transactional-Execution Facility support
HHC01417I With    "Optimized" instructions
HHC01417I With    OPTION_USE_SKAIP_AS_LOCK
HHC01417I With    OPTION_SIE2BK_FLD_COPY
HHC01417I With    OPTION_IODELAY_KLUDGE
HHC01417I With    OPTION_MVS_TELNET_WORKAROUND
HHC01417I With    OPTION_SIE_PURGE_DAT_ALWAYS
HHC01417I With    OPTION_NOASYNC_SF_CMDS
HHC01417I Machine dependent assists: cmpxchg1 cmpxchg4 cmpxchg8 cmpxchg16 hatomics=C11
HHC01417I Running on: zosatl100 (Linux-6.1.0-27-amd64 x86_64) MP=8
HHC01417I Built with crypto external package version 1.0.0.52-ga5096e5
HHC01417I Built with decNumber external package version 3.68.0.102-g3aa2f45
HHC01417I Built with SoftFloat external package version 3.5.0.106-gc114c53
HHC01417I Built with telnet external package version 1.0.0.63-g729f0b6
```

This was seen on Debian 12 and Rocky 9

**Comment by atncsj6h** [comment]
do you realize that port 23 is the default tcp ip port of the operating system ?
try using a different port number above 1023

`Port numbers from 0 to 1023 are reserved for common TCP/IP applications and are called well-known ports. The use of well-known ports allows client applications to easily locate the corresponding server application processes on other hosts.
`

**Comment by atncsj6h** [comment]
good luck!

**Comment by atncsj6h** [comment]
post your configuration file
the full log of the run
and describe the step and the data you use to connect to hercules

**Comment by miguemely** [comment]
@atncsj6h This can be reproduced even with defining no config file on run of Hercules and manually defining the CNSLPORT.

Not sure if the config is still needed. Here's the output of a full run. No matter which method I would try to use to connect to hercules, none will work since netstat doesn't show me that hercules is listening on a port.

Full output:
```
HHC01413I Hercules version 4.8.0.11356-SDL-DEV-ga215caf4
HHC01414I (C) Copyright 1999-2024 by Roger Bowler, Jan Jaeger, and others
HHC01417I Built for you with Hercules-Helper (version: -unknown-)
HHC01415I Build date: Nov 22 2024 at 13:43:52
HHC01417I Built with: GCC 12.2.0
HHC01417I Build type: GNU/Linux x86_64 host architecture build
HHC01417I Modes: S/370 ESA/390 z/Arch
HHC01417I Max CPU Engines: 128
HHC01417I Using   shared libraries
HHC01417I Using   setresuid() for setting privileges
HHC01417I Using   POSIX threads Threading Model
HHC01417I Using   Error-Checking Mutex Locking Model
HHC01417I With    Shared Devices support
HHC01417I With    Dynamic loading support
HHC01417I With    External GUI support
HHC01417I With    IPV6 support
HHC01417I With    HTTP Server support
HHC01417I With    sqrtl support
HHC01417I With    Signal handling
HHC01417I With    Watchdog monitoring
HHC01417I With    CCKD BZIP2 support
HHC01417I With    HET BZIP2 support
HHC01417I With    ZLIB support
HHC01417I With    Regular Expressions support
HHC01417I Without Object REXX support
HHC01417I With    Regina REXX support
HHC01417I With    Automatic Operator support
HHC01417I Without National Language Support
HHC01417I With    CCKD64 Support
HHC01417I With    Transactional-Execution Facility support
HHC01417I With    "Optimized" instructions
HHC01417I With    OPTION_USE_SKAIP_AS_LOCK
HHC01417I With    OPTION_SIE2BK_FLD_COPY
HHC01417I With    OPTION_IODELAY_KLUDGE
HHC01417I With    OPTION_MVS_TELNET_WORKAROUND
HHC01417I With    OPTION_SIE_PURGE_DAT_ALWAYS
HHC01417I With    OPTION_NOASYNC_SF_CMDS
HHC01417I Machine dependent assists: cmpxchg1 cmpxchg4 cmpxchg8 cmpxchg16 hatomics=C11
HHC01417I Running on: zosatl100 (Linux-6.1.0-27-amd64 x86_64) MP=8
HHC01417I Built with crypto external package version 1.0.0.52-ga5096e5
HHC01417I Built with decNumber external package version 3.68.0.102-g3aa2f45
HHC01417I Built with SoftFloat external package version 3.5.0.106-gc114c53
HHC01417I Built with telnet external package version 1.0.0.63-g729f0b6
HHC00100I Thread id 00007f5247ac0b80, prio 5, name 'impl_thread' started
HHC00100I Thread id 00007f5246e3e6c0, prio 4, name 'logger_thread' started
HHC00017W Crash dumps NOT enabled
HHC00007I Previous message from function 'impl' at impl.c(1376)
HHC00018I Hercules is running in elevated mode
HHC00150I Crypto module loaded (C) Copyright 2003-2024 by Bernard van der Helm & Juergen Winkelmann
HHC00151I Activated facility: Message Security Assist
HHC00151I Activated facility: Message Security Assist Extension 1, 2, 3, 4, 5 and 7
HHC17528I REXX(Regina) VERSION: REXX-Regina_3.6(MT) 5.00 31 Dec 2011
HHC17529I REXX(Regina) SOURCE:  UNIX
HHC17525I REXX(Regina) Rexx has been started/enabled
HHC17500I REXX(Regina) Mode            : Command
HHC17500I REXX(Regina) MsgLevel        : Off
HHC17500I REXX(Regina) MsgPrefix       : Off
HHC17500I REXX(Regina) ErrPrefix       : Off
HHC17500I REXX(Regina) Resolver        : On
HHC17500I REXX(Regina) SysPath    ( 6) : On
HHC17500I REXX(Regina) RexxPath   ( 0) :
HHC17500I REXX(Regina) Extensions ( 8) : .REXX:.rexx:.REX:.rex:.CMD:.cmd:.RX:.rx
HHC00111I Thread CPU Time IS available (_POSIX_THREAD_CPUTIME=0)
HHC00100I Thread id 00007f52468f76c0, prio 2, name 'Processor CP00' started
HHC00811I Processor CP00: architecture mode z/Arch
HHC00100I Thread id 00007f52467f66c0, prio 7, name 'timer_thread' started
HHC02385W CPUMODEL 0586 does not technically support TXF
HHC00007I Previous message from function 'txf_model_warning' at transact.c(2562)
HHC17736W TXF: TIMERINT 50 is too small; using default of 400 instead
HHC00007I Previous message from function 'txf_set_timerint' at transact.c(2587)
HHC01417I With    sqrtl support                                                                                                                                                              +
HHC01417I With    Signal handling
HHC01417I With    Watchdog monitoring
HHC01417I With    CCKD BZIP2 support
HHC01417I With    HET BZIP2 support
HHC01417I With    ZLIB support
HHC01417I With    Regular Expressions support
HHC01417I Without Object REXX support
HHC01417I With    Regina REXX support
HHC01417I With    Automatic Operator support
HHC01417I Without National Language Support
HHC01417I With    CCKD64 Support
HHC01417I With    Transactional-Execution Facility support
HHC01417I With    "Optimized" instructions
HHC01417I With    OPTION_USE_SKAIP_AS_LOCK
HHC01417I With    OPTION_SIE2BK_FLD_COPY
HHC01417I With    OPTION_IODELAY_KLUDGE
HHC01417I With    OPTION_MVS_TELNET_WORKAROUND
HHC01417I With    OPTION_SIE_PURGE_DAT_ALWAYS
HHC01417I With    OPTION_NOASYNC_SF_CMDS
HHC01417I Machine dependent assists: cmpxchg1 cmpxchg4 cmpxchg8 cmpxchg16 hatomics=C11
HHC01417I Running on: zosatl100 (Linux-6.1.0-27-amd64 x86_64) MP=8
HHC01417I Built with crypto external package version 1.0.0.52-ga5096e5
HHC01417I Built with decNumber external package version 3.68.0.102-g3aa2f45
HHC01417I Built with SoftFloat external package version 3.5.0.106-gc114c53
HHC01417I Built with telnet external package version 1.0.0.63-g729f0b6
HHC00100I Thread id 00007f5247ac0b80, prio 5, name 'impl_thread' started
HHC00100I Thread id 00007f5246e3e6c0, prio 4, name 'logger_thread' started
HHC00017W Crash dumps NOT enabled
HHC00007I Previous message from function 'impl' at impl.c(1376)
HHC00018I Hercules is running in elevated mode
HHC00150I Crypto module loaded (C) Copyright 2003-2024 by Bernard van der Helm & Juergen Winkelmann
HHC00151I Activated facility: Message Security Assist
HHC00151I Activated facility: Message Security Assist Extension 1, 2, 3, 4, 5 and 7
HHC17528I REXX(Regina) VERSION: REXX-Regina_3.6(MT) 5.00 31 Dec 2011
HHC17529I REXX(Regina) SOURCE:  UNIX
HHC17525I REXX(Regina) Rexx has been started/enabled
HHC17500I REXX(Regina) Mode            : Command
HHC17500I REXX(Regina) MsgLevel        : Off
HHC17500I REXX(Regina) MsgPrefix       : Off
HHC17500I REXX(Regina) ErrPrefix       : Off
HHC17500I REXX(Regina) Resolver        : On
HHC17500I REXX(Regina) SysPath    ( 6) : On
HHC17500I REXX(Regina) RexxPath   ( 0) :
HHC17500I REXX(Regina) Extensions ( 8) : .REXX:.rexx:.REX:.rex:.CMD:.cmd:.RX:.rx
HHC00111I Thread CPU Time IS available (_POSIX_THREAD_CPUTIME=0)
HHC00100I Thread id 00007f52468f76c0, prio 2, name 'Processor CP00' started
HHC00811I Processor CP00: architecture mode z/Arch
HHC00100I Thread id 00007f52467f66c0, prio 7, name 'timer_thread' started
HHC02385W CPUMODEL 0586 does not technically support TXF
HHC00007I Previous message from function 'txf_model_warning' at transact.c(2562)
HHC17736W TXF: TIMERINT 50 is too small; using default of 400 instead
HHC00007I Previous message from function 'txf_set_timerint' at transact.c(2587)
HHC00100I Thread id 00007f5245ef36c0, prio 2, name 'rubato_thread' started
HHC01603I CNSLPORT 3270
HHC02204I CNSLPORT       set to 3270
```

Here we see that we don't see a LISTEN for hercules.
![image](https://github.com/user-attachments/assets/c846ea7c-2942-4b3c-9356-b0b4e0c1d60c)

**Comment by atncsj6h** [comment]
> Not sure if the config is still needed.

**if I asked for it then it is needed**

started hercules without config file..

HHC01603I cnslport                                                                                                      
HHC17001I Console server listening on port 3270                                                                         
HHC01603I attach 01f 3270                                                                                               
HHC00100I Thread id 000000016ddcb000, prio 16, name 'console_connect' started                                           
HHC01024I Waiting for console connections on port 3270                                                                  
HHC02915I client 0 COMM: Connection received                                                                            
HHC02914I 0:001F COMM: client 0 negotiations complete; ttype = 'IBM-3279-3-E'                                           
HHC01018I 0:001F COMM: client 127.0.0.1 devtype 3270: connected                                                         
herc =====>                                                          

before the attach the 3270 emulator showed nothing 

after the attach the 3270 emulator screen shows
 Hercules Version  : 4.8.0.11346-SDL-DEV-gdc03da80-modified                    
 Host name         : enrico-mbp.local                                          
 Host OS           : Darwin-24.1.0                                             
 Host Architecture : arm64                                                     
 Processors        : LP=12, Cores=12, CPUs=1                                   
 LPAR Name         : HERCULES                                                  
 Device number     : 0:001F                                                    
 Subchannel        : 0000                                                      
                                                                               
          The Hercules S/370, ESA/390 and z/Architecture Emulator              
                           SDL 4.x Hyperion                                    
            _    _                            _                                
           ¦ ¦  ¦ ¦                          (_)                               
           ¦ ¦__¦ ¦ _   _  _ __    ___  _ __  _   ___   _ __                   
           ¦  __  ¦¦ ¦ ¦ ¦¦ '_ \  / _ \¦ '__¦¦ ¦ / _ \ ¦ '_ \                  
           ¦ ¦  ¦ ¦¦ ¦_¦ ¦¦ ¦_) ¦¦  __/¦ ¦   ¦ ¦¦ (_) ¦¦ ¦ ¦ ¦                 
           ¦_¦  ¦_¦ \__, ¦¦ .__/  \___¦¦_¦   ¦_¦ \___/ ¦_¦ ¦_¦                 
                     __/ ¦¦ ¦                                                  
                    ¦___/ ¦_¦                                                  
                                                                               
                    My PC thinks it's a MAINFRAME!                             
                                                                               
        Copyright (C) 1999-2024 Roger Bowler, Jan Jaeger, and others

**Comment by atncsj6h** [comment]
the  fact that message  
HHC01024I Waiting for console connections on port xxxx
does not show in your execution log hints
that you have no 3270 devices define in your configuration

**Comment by miguemely** [comment]
So then let me confirm here, if `HHC17001I Console server listening on port 3270` is shown, hercules is not actually listening on 3720, and you still need to define a connection (like `attach 01f 3270`?)

Config:

```
ARCHLVL  z/ARCH 
FACILITY ENABLE 006_ASN_LX_REUSE
FACILITY ENABLE 044_PFPO 
FACILITY ENABLE 080_DFP_PACK_CONV
FACILITY ENABLE 037_FP_EXTENSION
FACILITY ENABLE 050_CONSTR_TRANSACT
FACILITY ENABLE 051_LOCAL_TLB_CLEARING 
FACILITY ENABLE 073_TRANSACT_EXEC
FACILITY ENABLE 074_STORE_HYPER_INFO
FACILITY ENABLE 145_INS_REF_BITS_MULT
CNSLPORT 3270
CODEPAGE  DEFAULT
CPUMODEL  8562	   # CPU model number - Z15
CPUSERIAL 02B7F8   # CPU serial number - Z15
MODEL     T02 S03  # STSI returned Hardware model Capacity model - Z15
CPUVERID  00
ENGINES   CP   
LOADPARM  DE28K2M.
LOG       /hercules/log/zos31.log
MODPATH   /hercules/test-debian/hyperion/
LOGOPT    TIMESTAMP
LPARNAME  VS01
MAINSIZE  12G
MANUFACTURER IBM
MAXCPU    8 
NUMCPU    6
OSTAILOR  z/OS
PGMPRDOS  LICENSED 
PLANT     02
SYSEPOCH  1900
TZOFFSET  -0300
TIMERINT 400
CONKPALV (3,1,10)
DE25 3390   /hercules/zos31_version1/CICS61.CCKD cu=3990-6
DE26 3390   /hercules/zos31_version1/D31VS1.CCKD cu=3990-6
DE27 3390   /hercules/zos31_version1/Z31VS1.CCKD cu=3990-6
DE28 3390   /hercules/zos31_version1/OPEVS1.CCKD cu=3990-6
DE29 3390   /hercules/zos31_version1/STGVS1.CCKD cu=3990-6
DE2A 3390   /hercules/zos31_version1/DEVVS1.CCKD cu=3990-6
DE2B 3390   /hercules/zos31_version1/IMSV15.CCKD cu=3990-6
DE2C 3390   /hercules/zos31_version1/JCKVS1.CCKD cu=3990-6
DE2D 3390   /hercules/zos31_version1/JSPVS1.CCKD cu=3990-6
DE2E 3390   /hercules/zos31_version1/MQCD93.CCKD cu=3990-6
DE2F 3390   /hercules/zos31_version1/PRDVS1.CCKD cu=3990-6
DE30 3390   /hercules/zos31_version1/DB2V13.CCKD cu=3990-6
DE31 3390   /hercules/zos31_version1/T31VS1.CCKD cu=3990-6
DE32 3390   /hercules/zos31_version1/ZFSVS1.CCKD cu=3990-6
DE33 3390   /hercules/zos31_version1/USRVS1.CCKD cu=3990-6
1500.3 QETH iface 18-C0-4D-F3-E8-7C ipaddr 192.168.100.150/24 netmask 255.255.255.0 hwaddr 02-00-00-11-10-00 mtu 1500 chpid F0
0060	3270
0061 	3270
```

**Comment by atncsj6h** [comment]
> HHC17001I Console server listening on port 3270 

the message tells only what is defined in the system configuration for the CNSLPORT

why should hercules watch/listen to  the CNSLPORT if there are no 3270 devices defined ?

**Comment by miguemely** [comment]
Then wouldn't it be better to change that message to "Console server port set to ____"?

**Comment by Fish-Git** [comment]
> Then wouldn't it be better to change that message to "Console server port set to ____"?

No.

The purpose of the `[CNSLPORT](https://sdl-hercules-390.github.io/html/hercconf.html#CNSLPORT)` (and `[SYSGPORT](https://sdl-hercules-390.github.io/html/hercconf.html#SYSGPORT)`) statements are to inform Hercules which port it should be listening for _locally attached_ 3270 device TN3270 client connections on. (Hercules is always the _TN3270 server_, whereas your pre-connected TN3270 _client_ emulator program is always the TN3270 _cilent_).

Which port your guest operating system uses to communicate with its own _remote_ 3270 devices with is an entirely different matter. It can use whatever port number it wants. That's under the control of your mainframe's configuration file _(usually via an e.g. `PORT 23 TCP TN3270 ; (Telnet Server)` statement in your e.g. `ADCD.Z31A.TCPPARMS(PROF2)` member (which for z/OS 3.1 is commented out (disabled) by default))_

The `CNSLPORT` (and `SYSGPORT`) statements are only used for defining your mainframe's _physically attached **LOCAL** 3270_ display terminal devices. They are for which ports that Hercules's internal 3270 device _server_ should use. The ports that your guest operating system (e.g. z/OS) is entirely up to itself.

Does that make sense to you now?

In any case, I am closing this GitHub Issue as "WON'T FIX: WORKING AS DESIGNED".

Feel free to continue to post comments in this thread if you need further clarification or enlightenment. Just because the issue is closed, you can still post comments to it.

I hope that Helps!

_(p.s. Thanks for helping Enrico!)_

**Comment by miguemely** [comment]
Hi Fish! Thanks for taking the time to reply!

>ADCD.Z31A.TCPPARMS(PROF2)
This issue had nothing to do with what the underlying emulated OS was doing. It is only for the local console connection.

So I want to make sure I get this correctly:

CONSOLEP1|    (using attach)                       (CNSLPORT)
                    | <-----------------> hercules <--------------> remote 3270 client
CONSOLEP2|

In this example, CONSOLEP1 and CONSOLEP2 are (for lack of a better example) connected via serial to Hercules (a terminal server), and the client will connect to hercules via the defined CNSLPORT.


The problem that stems here is that if someone reads the documentation, it makes it sound like you only need to do `CNSLPORT (insertporthere)` and Hercules will start listening, even without defining any attached local 3720 sessions. The log should read better something like this:


```
HHC01603I CNSLPORT 3270
HHC17001I ~~Console server listening on port 3270~~ Console server port set to 3270
HHC01603I attach 01f 3270
HHC00100I Thread id 000000016ddcb000, prio 16, name 'console_connect' started
HHC01024I Waiting for console connections on port 3270
```

In most linux applications, if the log says that "something" is listening on a port, it should already have the port open and viewable in something like `netstat`, otherwise, the System Administrator will start to check things that might not be relevant `selinux, appguard, etc, etc`

Funny enough, I think in the newer version I built, it does exactly what I described for the logs (albeit a bit simpler), which is 

```
HHC01603I CNSLPORT 3270
HHC02204I CNSLPORT       set to 3270
```


But ^all of that^ actually throws off on a tangent.


Original issue was the config I provided has a 3270 defined:

```
0060	3270
0061 	3270
```
Which I've come to figure out, was the spacing was an issue here. 

```
0060 3270
0061 3270
```

**Comment by Fish-Git** [comment]
> So I want to make sure I get this correctly:
> 
> CONSOLEP1| (using attach) (CNSLPORT) | <-----------------> hercules <--------------> remote 3270 client CONSOLEP2|
> 
> In this example, CONSOLEP1 and CONSOLEP2 are (for lack of a better example) connected via serial to Hercules (a terminal server), and the client will connect to hercules via the defined CNSLPORT.

Correct.

> The problem that stems here is that if someone reads the documentation, it makes it sound like you only need to do `CNSLPORT (insertporthere)` and Hercules will start listening, even without defining any attached local 3720 sessions.

Ah. Yes. I think I see your problem now.  Our documentation is missing clarification that a `CNSLPORT` statement is only needed (is only used) if your configuration defines any 3270 devices. That if you don't have any 3270 devices defined in your configuration, then a `CNSLPORT` statement is no needed and will never be used. Yes?

Well, then I shall go ahead and change the status of your issue back to "BUG" _(or perhaps "ENHANCEMENT"?)_ and "DOC" and "IN PROGRESS"! Cool? &nbsp;`:)`

> The log should read better something like this:
> <pre>
> HHC01603I CNSLPORT 3270
> HHC17001I <strike> Console server listening on port 3270</strike> Console server port set to 3270
> <b>HHC01603I attach 01f 3270</b>
> HHC00100I Thread id 000000016ddcb000, prio 16, name 'console_connect' started
> <b>HHC01024I Waiting for console connections on port 3270</b>
> </pre>

Fair enough. I will make that change too!

> In most linux applications, if the log says that "something" is listening on a port, it should already have the port open and viewable in something like `netstat`, otherwise, the System Administrator will start to check things that might not be relevant `selinux, appguard, etc, etc`

I understand. But in this case `netstat` et al. will only show port number _\<cnslport\>_ as listening for connections if you have at least _one or more_ 3270 devices defined in your Hercules configuration. If you don't have any 3270 devices defined in your configuration file, then there is _no reason_ for Hercules to be listening for any incoming connections on that port.

> But _all of that_ actually throws off on a tangent from the original issue.
> 
> Original issue was the config I provided has a 3270 defined incorrectly:
> 
> ```
> 0060	3270
> 0061 	3270
> ```

Huh? There's nothing wrong with those two statements.

> Which I've come to figure out, was the spacing was an issue here.

Spacing?

Again, the amount space between the device number and its type should not matter, so I'm a little confused by your claim. What are you referring to? Did you experience some type of problem when multiple spaces were used between the device number and device type? Id so, then please attach your Hercules configuration and Log files so I can see what the heck is going on! Thanks.

> ```
> 0060 3270
> 0061 3270
> ```

Which should be _exactly_ equivalent to your previous pair of device statements!

> But because of the documentation, I was lead to believe it was something to do with the CNSLPORT.

Ah! Yes. I see. It's the fact that we chose our default _listening port number_ to be _(in this particular instance)_ the same as the device's device type. Yes, I can see how that might mislead and/or confuse someone. Unfortunately however, I can tell you that _THAT_ is not ever going to change. It's original legacy behavior and we are naturally leary to be suddenly and unexpectedly Hercules defauly behavior. We can add support to provoke _new_ (different) behavior behavior, but we are loathe to ever change existing pre-established behavior due to the unwanted "surprise" factor. I'm sure you understand.

If there is some wording in our documentation that you feel could be made clearer and less misleading however, we would certainly appreciate you identifying such areas and such rewordings so we can get it fixed. Documentation is mostly for our users after all, and we want it to be as clear as possible. Unclear or confusing or misleading User Documentation is bad as it can lead to a confusing and very annoying user experience, which we definitely don't want to happen!

Cool?

**Comment by miguemely** [comment]
> Spacing?

That's what I said! Internally, in my head, it made no sense, because I've seen other configs with spacing before. I'll get you the logs. The config I actually provided up here (https://github.com/SDL-Hercules-390/hyperion/issues/692#issuecomment-2495548302)

**Comment by Fish-Git** [comment]
Fixed(?) (HOPEFULLY! You tell me!) by commit 3de32a797c389035295d0444f57825aeae2ca7ce.

Closing.

**Comment by miguemely** [comment]
That looks great. Thank you very much! This is my first foray into mainframes, so I wanted to make it so other people walking into it could do it as well!

Thank you for maintaining this project!
